database/factories/ArticleFactory.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Database\Factories;
+
+use App\Models\Article;
+use App\Models\User;
+use Database\Factories\Traits\RandomTimestamps;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+/**
+ * @extends Factory<Article>
+ */
+class ArticleFactory extends Factory
+{
+    use RandomTimestamps;
+
+    /**
+     * Define the model's default state.
+     *
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        return [
+            'title' => fake()->name(),
+            'category' => fake()->name(),
+            'published_at' => fake()->date(),
+            'meta_description' => fake()->text(),
+            'content' => fake()->text(),
+            'user_id' => User::factory()->withWallet(),
```

Comment:
```suggestion
            'user_id' => fn () => User::factory()->withWallet(),
```

so we can pass it a user as well

Reply:
think you can still override the `user_id`, if you pass the factory instance it will lazily create it only if you don't manually set `user_id`

Reply:
ah indeed, I was under the impression it required `fn()` for that